zebra/zebra_vxlan.c

```diff
@@ -5159,6 +5159,15 @@ void zebra_vxlan_macvlan_up(struct interface *ifp)
 	zif = ifp->info;
 	assert(zif);
 	link_ifp = zif->link;
+	if (!link_ifp) {
```

Comment:
In https://github.com/FRRouting/frr/issues/13523 only FRR configuration is provided. What are the interfaces ? Does the macvlan eventually comes up? There are asserts added for reason as this is not expected.

The API only called when ifp is type MACVLAN. Have you checked during interface down was zif cleaned up?
```
1044     } else if (IS_ZEBRA_IF_MACVLAN(ifp)) { 
1045         zebra_vxlan_macvlan_up(ifp);
1046     }

Reply:
Well too bad nobody documented the reason why the asserts were added. The interfaces are just the parent interface in the main network namespace and then macvlans in the non-main network namespace. I think the zif->link is always NULL, this is just the place it is accessed without checks.

Reply:
@tlsalmin Zebra can watch netns mounts in `/run/netns` and handle rtnl events in all net namespaces. The problem arises when it can't access that directory (e.g. in another mount ns set up by docker) so can't find the parent interface.

Reply:
In this use case this instance of zebra in a specific namespace shouldn't have access to any other namespace.

Reply:
In this use case this instance of zebra in a specific namespace shouldn't have access to any other namespace.